lib/functions/general/apt-utils.sh

```diff
@@ -41,35 +40,29 @@ function apt_find_upstream_package_version_and_download_url() {
 
 	declare base_down_url="http://${mirror_with_slash}pool/main/${first_letter_of_sought_package_name}/${sought_package_name}"
 
-	declare index package_info_download_url
-	# loop over the package_info_download_urls with index and value
-	for index in "${!package_info_download_urls[@]}"; do
-		package_info_download_url="${package_info_download_urls[$index]}"
-		display_alert "Testing URL" "${package_info_download_url}" "debug"
-
-		declare package_info_download_url_file
-		package_info_download_url_file="$(mktemp)"
-		curl --silent --show-error --max-time 10 "${package_info_download_url}" > "${package_info_download_url_file}" || true # don't fail
-		declare package_info_download_url_file_package_name                                                                   # grep the file for the package name. parse "<kbd>name</kbd>"
-		package_info_download_url_file_package_name="$(grep -oP '(?<=<kbd>)[^<]+' "${package_info_download_url_file}" | grep "^${sought_package_name}_" | head -n 1)"
-		rm -f "${package_info_download_url_file}"
-
-		display_alert "Package name parsed" "${package_info_download_url_file_package_name}" "debug"
-		if [[ "${package_info_download_url_file_package_name}" == "${sought_package_name}_"* ]]; then
-			found_package_filename="${package_info_download_url_file_package_name}"
-			found_package_down_url="${base_down_url}/${found_package_filename}"
-			display_alert "Found package filename" "${found_package_filename} in url ${package_info_download_url}" "debug"
-			break
-		fi
-	done
+	# get json with latest pacakge info generated by GHA
+	case "${GITHUB_MIRROR}" in
+		"ghproxy")
+			package_info_download_url="https://ghp.ci/https://raw.githubusercontent.com/armbian/armbian.github.io/refs/heads/data/data/${sought_package_name}.json"
```

Comment:
This proxy is limited to repository content, so we can't use nice URL's. Is there a way?

Reply:
Reverse proxy like ghproxy does not support github pages. We can use https://github.armbian.com/base-files.json, and at least at the moment the url is reachable from China.

Reply:
But in case github pages is down for some reason, the ghproxy should save the world for user, it's better to keep this fall back.

Reply:
> Reverse proxy like ghproxy does not support github pages.

Got it. Now which path to go:
- keep it as proposed
- remove ghproxy and we wait and see

Reply:
I recommend keeping it.